src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -180,17 +182,21 @@ private[health] class HealthCheckActor(
             case None =>
               log.error(s"Couldn't find instance $instanceId")
               health.update(result)
-          }
+          })
       }
 
-      healthByInstanceId += (instanceId -> newHealth)
+      futureHealth.onComplete {
+        case Success(newHealth) =>
+          healthByInstanceId += (instanceId -> newHealth)
```

Comment:
pretty sure that this is an unsafe operation now that it's completing in the context of a future instead of an actor mailbox. consider changing the type of `healthByInstanceId` to a TrieMap